lib/create-contentful-api.ts

```diff
@@ -122,12 +123,22 @@ export default function createContentfulApi<OptionType extends ChainOptions>(
     return query
   }
 
+  function maybeEncodeCPAResponse(data: any, config: Record<string, any>): any {
+    const includeContentSourceMaps = config?.query?.includeContentSourceMaps as boolean
+
+    if (includeContentSourceMaps) {
+      return encodeCPAResponse(data)
+    }
+
+    return data
+  }
+
   async function get<T>({ context, path, config }: GetConfig): Promise<T> {
     const baseUrl = getBaseUrl(context)
 
     try {
       const response = await http.get(baseUrl + path, config)
-      return response.data
+      return maybeEncodeCPAResponse(response.data, config)
```

Comment:
Are we sure this only applies to responses from entries and assets endpoints?

Reply:
yes `includeContentSourceMaps` is only set to true when we use maybeEnableSourceMaps like [this](https://github.com/contentful/contentful.js/blob/980b087f5bd0381eb6b1c289b53e55c37d049f1e/lib/create-contentful-api.ts#L260) & we have only added it to the entries and assets endpoints